system-addon/common/ShortURL.jsm

```diff
@@ -15,18 +18,34 @@ Components.utils.importGlobalProperties(["URL"]);
  *         {str} link.title (optional) - The title of the link
  * @return {str}   A short url
  */
-this.shortURL = function shortURL(link) {
-  if (!link.url && !link.hostname) {
-    return "";
+this.ShortURL = class ShortURL {
+  constructor() {
+    this.IDNService = Cc["@mozilla.org/network/idn-service;1"].getService(Ci.nsIIDNService);
+  }
+
+  getEffectiveHost(url) {
```

Comment:
A comment block for the method and renaming the parameter to `host` would make more sense than `url`. Also, what does "effective host" mean?